Neos.RedirectHandler/Classes/Command/RedirectionCommandController.php

```diff
@@ -0,0 +1,342 @@
+<?php
+namespace Neos\RedirectHandler\Command;
+
+/*
+ * This file is part of the TYPO3.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use League\Csv\Reader;
+use League\Csv\Writer;
+use Neos\RedirectHandler\Exception;
+use Neos\RedirectHandler\RedirectInterface;
+use Neos\RedirectHandler\Storage\RedirectStorageInterface;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Cli\CommandController;
+use TYPO3\Flow\Persistence\PersistenceManagerInterface;
+use TYPO3\Flow\Utility\Arrays;
+
+/**
+ * Command controller for tasks related to redirects
+ *
+ * @Flow\Scope("singleton")
+ */
+class RedirectionCommandController extends CommandController
+{
+    /**
+     * @Flow\Inject
+     * @var RedirectStorageInterface
+     */
+    protected $redirectStorage;
+
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @param string $host Filter redirects by the given hostname
+     * @param string $match Match source or target URI by a regular expression
+     * @return void
+     */
+    public function listCommand($host = null, $match = null)
+    {
+        $outputByHost = function ($host = null) use ($match) {
+            $redirects = $this->redirectStorage->getAll($host);
+            $this->outputLine();
+            if ($host !== null) {
+                $this->outputLine('<info>==</info> <b>Redirect for %s</b>', [$host]);
+            } else {
+                $this->outputLine('<info>==</info> <b>Redirect without host attached</b>', [$host]);
```

Comment:
Maybe say "Redirects valid for all hosts" (underwise this could be misunterstood)